### PR TITLE
Feat/skybox domes shader

### PR DESCRIPTION
**Pull request by dcl-glb**
## What does this PR change?

Skybox dome integration
## How to test the changes?

1. https://play.decentraland.zone/?renderer=https://glob-dland-s3-unity-renderer.s3.amazonaws.com/branch/feat/Skybox-DomesShader

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

File: unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/BaseLayer.cs [hunk 1]
```diff
@@ -100,6 +101,127 @@ public TextureLayer(string name = "noname")
 
             particleRotation = new List<TransitioningVector3>();
         }
+
+        // Copy details of present layer on the passed layer
+        public void DeepCopy(TextureLayer layer)
+        {
+            layer.slotID = this.slotID;
```
Comment (BrianAmadori) [inline unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/BaseLayer.cs]: Can we use `MemberwiseClone` for primitives and only do the deepcopy for reference types? This approach will lower the high maintenance cost of this method.

File: unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/Editor/Procedural Skybox/SkyboxEditor_v2.cs [hunk 2]
```diff
@@ -559,6 +497,14 @@ private void EnsureDependencies()
                 directionalLight = temp.AddComponent<Light>();
                 directionalLight.type = LightType.Directional;
             }
+
+            // Init 3D
+            //Init3DSetup();
```
Comment (BrianAmadori) [inline unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/Editor/Procedural Skybox/SkyboxEditor_v2.cs]:
```suggestion
```

File: unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/Editor/Procedural Skybox/SkyboxEditor_v2.cs [hunk 3]
```diff
@@ -559,6 +497,14 @@ private void EnsureDependencies()
...
+            if (skyboxObjects == null)
+            {
+                skyboxObjects = new SkyboxGameobjectsPool();
```
Comment (BrianAmadori) [inline unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/Editor/Procedural Skybox/SkyboxEditor_v2.cs]:
```suggestion
                skyboxObjects = new SkyboxGameObjectsPool();
```

File: unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/SkyboxGameobjectsPool.cs [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+                    Debug.Log("Making Skybox elements");
+                    skyboxElements = new GameObject("Skybox Elements");
+                    skyboxElements.layer = LayerMask.NameToLayer("Skybox");
+                    domeElements = new GameObject("Dome Elements");
+                    domeElements.layer = LayerMask.NameToLayer("Skybox");
+                    domeElements.transform.parent = skyboxElements.transform;
+                }
+                else
+                {
+                    Debug.Log("Fetching Skybox elements");
```
Comment (BrianAmadori) [inline unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/SkyboxGameobjectsPool.cs]:
```suggestion
```

File: unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/SkyboxGameobjectsPool.cs [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+                dome = InstantiateNewDome();
+            }
+            else
+            {
+                dome = domeObjects.Dequeue();
+            }
+
+            dome.domeGO.SetActive(true);
+            // Resize dome object
+            //dome.domeGO.transform.localScale = dome.domeGO.transform.localScale + Vector3.one * numberOf3DObjects;
```
Comment (BrianAmadori) [inline unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/SkyboxGameobjectsPool.cs]:
```suggestion
```

Please remember to remove all commented code before submitting for review

File: unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/BaseLayer.cs (hunk 1 above)
Comment (dcl-glb) [reply unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/BaseLayer.cs]: done

**Pull request closed without merging**